### ISSUE TITLE
Encryption at rest not supported by OrientGraphFactory

### ISSUE THREAD

**Issue by odbuser2** [orientechnologies/orientdb#6231]
## Expected behavior and actual behavior
Expected : OrientGraphFactory should contain methods/arguments to support encryption method/key like ODatabaseDocumentTx.

Actual : OrientGraphFactory does not contain methods/arguments to support encryption method/key

## Steps to reproduce the problem
Just look at the OrientGraphFactory creation options.  The closest support is to use:
`public OrientGraphFactory(String iURL, String iUser, String iPassword, OPartitionedDatabasePool pool)
`
and supply a custom OPartitionedDatabasePool however extending this class will not work.  You could extend it and override:
`protected void openDatabase(final DatabaseDocumentTxPolled db)`
and create your own DatabaseDocumentTxPolled but DatabaseDocumentTxPolled is private which prevents you from overriding openDatabase.

e.g. you can't do this:
```
OrientGraphFactory oGF = new OrientGraphFactory(url, user, pword, 
        new OPartitionedDatabasePool(url, user, pword) {
    @Override
    protected void openDatabase(OPartitionedDatabasePool.DatabaseDocumentTxPolled db) {
        db.setProperty(OGlobalConfiguration.STORAGE_ENCRYPTION_METHOD.getKey(), method);
        db.setProperty(OGlobalConfiguration.STORAGE_ENCRYPTION_KEY.getKey(), key);        
        super.openDatabase(db);
    }
});
```

Even so, it would be better to be able to set these properties via the constructor or builder/fluent design of OrientGraphFactory before the db is opened.

### Runninng Mode
- [x] Embedded, using PLOCAL access mode
- [ ] Embedded, using MEMORY access mode
- [ ] Remote

### Misc
- [ ] I have a distributed setup with multiple servers. How many? 
- [ ] I'm using the Enterprise Edition

### OrientDB Version
- [ ] v2.0.x - Please specify last number: 
- [ ] v2.1.x - Please specify last number: 
- [x] v2.2.x - Please specify last number: 

### Operating System
- [ ] Linux
- [x] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

### Java Version
- [ ] 6
- [ ] 7
- [x] 8

**Comment by lvca** [comment]
@robfrank can you take this?

**Comment by robfrank** [comment]
hi, I propagated encryption to the pool and to the OrientGraphFactory.
```
    OrientGraphFactory graphFactory = new OrientGraphFactory("plocal:" + dbPath);

    graphFactory.setProperty(STORAGE_ENCRYPTION_METHOD.getKey(), "des");
    graphFactory.setProperty(STORAGE_ENCRYPTION_KEY.getKey(), "T1JJRU5UREJfSVNfQ09PTA==");

    ODatabaseDocumentTx db = graphFactory.getDatabase();

```

it will be included on 2.2.6

**Comment by lvca** [comment]
Reverted because caused all distributed tests to fail.
